include/wil/result_macros.h

```diff
@@ -2851,8 +2851,8 @@ namespace details
     // NOTE: The following two functions are unfortunate copies of strsafe.h functions that have been copied to reduce the friction associated with using
     // Result.h and ResultException.h in a build that does not have WINAPI_PARTITION_DESKTOP defined (where these are conditionally enabled).
 
-    static STRSAFEAPI WilStringLengthWorkerA(
-        _In_reads_or_z_(cchMax) STRSAFE_PCNZCH psz,
+    inline HRESULT WilStringLengthWorkerA(

```

Comment:
Note this isn't quite a 1:1 change since `STRSAFEAPI` is defined as `__stdcall`. Since it's an inline definition, I don't think it should matter, though

Reply:
This could potentially be problematic if anyone is taking a function pointer to one of the functions, _however_
1. All instances appear to be in the `details` namespace, so no 3rd party code should be referencing these functions
2. I did a search on our internal code and saw no evidence of this potentially being done

So this should be fine. Do you know the reason why `STRSAFEAPI` caused issues in the first place in case I overlooked something and `__stdcall` needs to be added back?

Reply:
I assumed it's because of the extern c, not the calling convention, but I didn't dive into it. I can check whether leaving `__stdcall` will work, but I thought it's better to just be consistent with the other functions, and potentially improve the performance as the calling convention can be optimized the compiler now. My thought was, as you said, that no external code should depend oh the functions under `details`.

Reply:
> but I thought it's better to just be consistent with the other functions

Yeah, I think that's the best course of action at the moment. If issues arise, I may need to re-introduce `__stdcall`, so I was just wondering if that would start breaking things again (and e.g. if we'd need to use something like a macro instead). But that's unlikely

Reply:
OK, so for reference, with `static STRSAFEAPI` (expands to `static _STRSAFE_EXTERN_C HRESULT WINAPI`):
```
In file included from file.cpp:61:
In file included from Compiler/include/wil/resource.h:15:
Compiler/include/wil/result_macros.h:2854:12: error: cannot combine with previous 'static' declaration specifier
 2854 |     static STRSAFEAPI  WilStringLengthWorkerA(
      |            ^
Compiler/include/strsafe.h:85:20: note: expanded from macro 'STRSAFEAPI'
   85 | #define STRSAFEAPI _STRSAFE_EXTERN_C __inline HRESULT WINAPI
      |                    ^
Compiler/include/strsafe.h:71:27: note: expanded from macro '_STRSAFE_EXTERN_C'
   71 | #define _STRSAFE_EXTERN_C extern "C"
      |                           ^
In file included from file.cpp:61:
In file included from Compiler/include/wil/resource.h:15:
Compiler/include/wil/result_macros.h:2854:12: error: expected unqualified-id
Compiler/include/strsafe.h:85:20: note: expanded from macro 'STRSAFEAPI'
   85 | #define STRSAFEAPI _STRSAFE_EXTERN_C __inline HRESULT WINAPI
      |                    ^
Compiler/include/strsafe.h:71:34: note: expanded from macro '_STRSAFE_EXTERN_C'
   71 | #define _STRSAFE_EXTERN_C extern "C"
      |                                  ^
In file included from file.cpp:61:
In file included from Compiler/include/wil/resource.h:15:
Compiler/include/wil/result_macros.h:2898:18: error: use of undeclared identifier 'WilStringLengthWorkerA'; did you mean 'StringLengthWorkerA'?
 2898 |             hr = WilStringLengthWorkerA(psz, cchMax, pcchLength);
      |                  ^
Compiler/include/strsafe.h:1734:12: note: 'StringLengthWorkerA' declared here
 1734 | STRSAFEAPI StringLengthWorkerA(STRSAFE_LPCSTR psz,size_t cchMax,size_t *pcchLength) {
      |            ^
3 errors generated.
```
Removing `_STRSAFE_EXTERN_C` (i.e. with `static HRESULT WINAPI`) works.